mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -620,10 +620,21 @@ LogicalResult gpu::SubgroupReduceOp::verify() {
                        << "` reduction operation is not compatible with type "
                        << getType();
   }
+
+  if (auto clusterSize = getClusterSize()) {
+    uint32_t size = *clusterSize;
+    if (!llvm::isPowerOf2_32(size)) {
+      return emitError() << "cluster size " << size << " is not a power of two";
```

Comment:
```suggestion
      return emitOpError() << "cluster size " << size << " is not a power of two";
```

To print the op inline with the error message

Reply:
I've made this change, but to be honest I'm not sure why this is better. It shows the line where the error happened either way, right?

Reply:
I find op errors to be easier to work with because in real models the line number loc doesn't really match anything useful (high level ops get decomposed into dozens/hundreds of simpler instructions all with the same location)

Reply:
I see, thanks.